packages/react-storage/src/components/StorageBrowser/Views/Controls/Navigate.tsx

```diff
@@ -93,23 +91,46 @@ export interface NavigateControl<
 }
 
 export const NavigateControl: NavigateControl = (_props) => {
+  const [state, handleUpdateState] = useControl({ type: 'NAVIGATE' });
+
   return (
     <NavigateContainer>
       <NavigateItem
-        item={{
-          name: 'Home',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-      />
-      <NavigateItem
-        item={{
-          name: 'Location01',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-        current
+        item={HOME_NAVIGATE_ITEM}
+        current={!state.location}
+        onClick={() => handleUpdateState({ type: 'DESELECT_LOCATION' })}
       />
+      {state.location && (
+        <>
+          <NavigateItem
+            item={state.location.bucket}
+            current={!state.history || state.history.length === 0}
+            onClick={() => {
+              if (state?.location) {
```

Comment:
Just so i understand correctly, is the optional chaining here just due to types or is it possible that a `NavigateItem` can be rendered that won't do anything on click?

Reply:
ah sorry, I just removed the optional chaining locally and it looks like it's not complaining. I do have to add the null check for `state.location` though because typescript says it could be undefined